phonenumbers.go

```diff
@@ -3412,20 +3411,24 @@ func GetTimezonesForNumber(number *PhoneNumber) ([]string, error) {
 	return GetTimezonesForPrefix(e164)
 }
 
-func getValueForNumber(onceMap map[string]*sync.Once, langMap map[string]*intStringMap, binMap map[string]string, language string, maxLength int, number *PhoneNumber) (string, int, error) {
+func getValueForNumber(visited *sync.Map, langMap map[string]*intStringMap, dir string, binMap embed.FS, language string, maxLength int, number *PhoneNumber) (string, int, error) {
 	// do we have data for this language
-	_, existing := binMap[language]
-	if !existing {
-		return "", 0, nil
+	file, err := binMap.ReadFile(dir + "/" + language + ".txt.gz")
+	if err != nil {
+		if errors.Is(err, fs.ErrNotExist) {
+			return "", 0, nil
+		}
+
+		return "", 0, err
 	}
 
 	// load it into our map
-	onceMap[language].Do(func() {
-		prefixMap, err := loadPrefixMap(binMap[language])
+	if _, loaded := visited.LoadOrStore(language, struct{}{}); !loaded {
```

Comment:
Doesn't this lead to a race condition if two threads get here at the same time looking for a language that hasn't yet been loaded.. first does the load, second doesn't.. but second could get to line 3434 before the first thread completes the loading?